### PR TITLE
[PS-1093] Move Organization Api Calls

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Move organization api methods out of `ApiService` into their own `OrganizationApiService`. This is part one of two. Part two will convert `OrganizationService` to providing observables. 

## Code changes

- Move organization api calls to `organization-api.service.ts`
- Remove usage of `any` in changed files

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

**Review by MGibson1 — CHANGES_REQUESTED** [review]

File: apps/desktop/src/app/accounts/set-password.component.ts [hunk 1]
```diff
@@ -16,6 +17,10 @@ import { SyncService } from "@bitwarden/common/abstractions/sync.service";
 
 const BroadcasterSubscriptionId = "SetPasswordComponent";
 
+interface Message {
+  command: string;
+}
+
```
[the remark this answers is not in the document]
Comment (justindbaur) [reply apps/desktop/src/app/accounts/set-password.component.ts]: Yep, good call. Added it!

File: apps/web/src/app/settings/adjust-payment.component.ts [hunk 2]
```diff
@@ -24,13 +25,14 @@ export class AdjustPaymentComponent {
   @Output() onCanceled = new EventEmitter();
 
   paymentMethodType = PaymentMethodType;
-  formPromise: Promise<any>;
+  formPromise: Promise<void | unknown>;
```
[the remark this answers is not in the document]
Comment (justindbaur) [reply apps/web/src/app/settings/adjust-payment.component.ts]: No, both the calls it holds a promise to don't actually return anything. One was returning `any` but I changed it to being `void`.

File: apps/web/src/app/settings/tax-info.component.ts [hunk 3]
```diff
@@ -3,10 +3,23 @@ import { ActivatedRoute } from "@angular/router";
...
+  taxId: string | null;
+  line1: string | null;
+  line2: string | null;
+  city: string | null;
+  state: string | null;
+  postalCode: string | null;
+  country: string;
+  includeTaxId: boolean;
+  [key: string]: unknown;
+}
```
[the remark this answers is not in the document]
Comment (justindbaur) [reply apps/web/src/app/settings/tax-info.component.ts]: Kinda, it's based on a response model but doesn't need some of the properties and has a few of it's own that are calculated based on other properties. I made it a `Omit` and union type to make it smaller and smarter.

File: apps/web/src/app/vault/vault-filter/organization-filter/organization-options.component.ts [hunk 4]
```diff
@@ -19,7 +20,7 @@ import { EnrollMasterPasswordReset } from "../../../organizations/users/enroll-m
   templateUrl: "organization-options.component.html",
 })
 export class OrganizationOptionsComponent {
-  actionPromise: Promise<any>;
+  actionPromise: Promise<boolean | unknown>;
```
[the remark this answers is not in the document]
Comment (justindbaur) [reply apps/web/src/app/vault/vault-filter/organization-filter/organization-options.component.ts]: Yes it's reused 3 times, twice for boolean and once for void (was `any`). I switched this to being a `boolean | void`.

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**